### PR TITLE
[15.0] connector_importer: misc fix/imp

### PR DESCRIPTION
**Pull request by simahawk**
See atomic commits for details.
Basically moving out of product importer work branch here https://github.com/camptocamp/connector-interfaces/tree/15-mig-connector_importer_product

Need to get back to this for review and tests.

### CODE REVIEW

**Review by dhx — APPROVED** [review]
Did some basic tests - LGTM

**Comment by simahawk** [comment]
pre-commit fails like

```
   File "/home/runner/.cache/pre-commit/repo3jub9f__/py_env-python3/lib/python3.11/site-packages/wrapt/__init__.py", line 10, in <module>
    from .decorators import (adapter_factory, AdapterFactory, decorator,
  File "/home/runner/.cache/pre-commit/repo3jub9f__/py_env-python3/lib/python3.11/site-packages/wrapt/decorators.py", line 34, in <module>
    from inspect import ismethod, isclass, formatargspec
ImportError: cannot import name 'formatargspec' from 'inspect' (/opt/hostedtoolcache/Python/3.11.0/x64/lib/python3.11/inspect.py)

pylint with mandatory checks.............................................Failed
- hook id: pylint
- exit code: 1
```
I'm trying a template sync here #97

File: connector_importer/components/mapper.py [hunk 1]
```diff
@@ -61,7 +66,23 @@ def translatable_keys(self, create=False):
 
         The recordset will use this to show translatable fields to users.
         """
-        return self.translatable
+        translatable = list(self.translatable)
+        translatable += self.work.options.mapper.get("translatable_keys", [])
+        translatable = self._validate_translate_keys(set(translatable))
+        return tuple(translatable)
+
+    def _validate_translate_keys(self, translatable):
```
Comment (sebalix) [inline connector_importer/components/mapper.py:74]: :+1:

File: connector_importer/models/recordset.py [hunk 2]
```diff
@@ -272,24 +271,21 @@ def generate_report(self):
...
     @api.depends("import_type_id")
     def _compute_docs_html(self):
         template = self.env.ref("connector_importer.recordset_docs")
         for item in self:
             item.docs_html = False
-            if isinstance(item.id, models.NewId):
+            if isinstance(item.id, models.NewId) or not item.backend_id:
+                # Surprise surprise: when editing a new recordset
+                # if you hit `configure source` btn
+                # the record will be saved but the backend can be null :S
```
Comment (sebalix) [inline connector_importer/models/recordset.py:288]: Not sure to understand, if you create a new recordset from the UI, as there is no `backend_id` on the form it is expected to not have it defined here no?

**Review by sebalix — APPROVED** [review]
Approving anyway, LG

**Pull request merged**